### PR TITLE
fix(curriculum): improve description and tests descriptions

### PR DESCRIPTION
**Pull request by gikf**
This ended up a bit more elaborate than expected.
- Clarifies a bit what will be input and what is expected output.
- Adds example.
- Adds, for consistency, sorting of output array with closest pair. Both in solution and in description.
- Adds third test, after adjusting existing third array with points to not have multiple point pairs with the same - closest - distance. Otherwise it would be needed to include in description how deal with such cases, This would make it even more complicated.
- Makes tests descriptions more explicit, including expected values.
- Uses MathJax to improve look of math notation.
- Removes external links, per #40749.
- For consistency in background code positions answer for each points array below of the array.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/closest-pair-problem.md [hunk 1]
```diff
@@ -65,16 +65,38 @@ A better algorithm is based on the recursive divide and conquer approach, as exp
...
+    },
+    {
+      x: 2,
+      y: 2
+    }
+  ]
+}
+```
+
+**Note:** In the `pair` array, sort poinits by the `x` coordinate, incrementing.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/closest-pair-problem.md:98]:
```suggestion
**Note:** Sort the `pair` array by their `x` values in incrementing order.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
Found one little typo and reworded that sentence. It doesn't have to be reworded to what I want - I thought it sounded a little better - but, the typo should be fixed. Other than that, LGTM 🎉 

I added this solution to the forum guide article

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/closest-pair-problem.md (hunk 1 above)
Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/closest-pair-problem.md:98]: Suggestion is definitely better :+1:

**Comment by moT01** [comment]
what

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Review by raisedadead — APPROVED** [review]

**Pull request merged**